frappe/core/doctype/user_invitation/user_invitation.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2025, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+import frappe
+import frappe.utils
+from frappe.model.document import Document
+
+
+class UserInvitation(Document):
+	# begin: auto-generated types
+	# This code is auto-generated. Do not modify anything in this block.
+
+	from typing import TYPE_CHECKING
+
+	if TYPE_CHECKING:
+		from frappe.types import DF
+
+		accepted_at: DF.Datetime | None
+		app_name: DF.Literal[None]
+		email: DF.Data
+		email_sent_at: DF.Datetime | None
+		invited_by: DF.Link | None
+		key: DF.Data | None
+		redirect_to_path: DF.Data
+		role: DF.Link
+		status: DF.Literal["Pending", "Accepted", "Expired"]
+		user: DF.Link | None
+	# end: auto-generated types
+
+	def before_insert(self):
+		frappe.utils.validate_email_address(self.email, throw=True)
+		self.key = frappe.generate_hash(length=12)
+		self.invited_by = frappe.session.user
+		self.status = "Pending"
+		if self.app_name is None:
+			self.app_name = "frappe"
+
+	def after_insert(self):
+		invite_link = frappe.utils.get_url(
+			f"/api/method/frappe.core.api.user_invitation.accept_invitation?key={self.key}"
+		)
+		email_title = self.get_email_title()
+		frappe.sendmail(
+			recipients=self.email,
+			subject=f"You've been invited to join {email_title}",
```

Comment:
https://docs.frappe.io/framework/user/en/translations 

Wrap this and other messages to allow translations.